src/Trie.mo

```diff
@@ -733,6 +743,41 @@ module {
     )
   };
 
+  /// Returns an `Iter` over the key-value entries of the trie.
+  ///
+  /// Each iterator gets a _persistent view_ of the mapping, independent of concurrent updates to the iterated map.
+  public func iter<K, V>(t : Trie<K, V>) : I.Iter<(K, V)> {
+    object {
+      var stack = ?(t, null) : List.List<Trie<K, V>>;
+      public func next() : ?(K, V) {
+        switch stack {
+        case null { null };
+        case (?(trie, stack2)) {
+               switch trie {
+               case (#empty) {
+                      stack := stack2;
+                      next()
+                    };
+               case (#leaf({keyvals = null})) {
+                      stack := stack2;
+                      next()
+                    };
+               case (#leaf({size = c; keyvals = ?((k, v), kvs)})) {
+                      stack := ?(#leaf({size=c-1; keyvals=kvs}), stack2);
+                      ?(k.key, v)
+                    };
+               case (#branch(br)) {
+                      stack := ?(br.left, ?(br.right, stack2));
+                      next()
+                    };
+               }
+             }
```

Comment:
```suggestion
      var stack = ?(t, null) : List.List<Trie<K, V>>;
      public func next() : ?(K, V) {
        switch stack {
          case null { null };
          case (?(trie, stack2)) {
              switch trie {
                 case (#empty) {
                     stack := stack2;
                     next()
                  };
                 case (#leaf({ keyvals = null })) {
                     stack := stack2;
                     next()
                 };
               case (#leaf({ size = c; keyvals = ?((k, v), kvs)})) {
                   stack := ?(#leaf({ size = c-1; keyvals = kvs}), stack2);
                   ?(k.key, v)
               };
               case (#branch(br)) {
                   stack := ?(br.left, ?(br.right, stack2));
                   next()
               };
             }
           }
```
something like that (not too sure about the record spacing - check what I did elsewhere I guess or consult the guidelines which I was trying to follow (modulo misunderstanding).

Reply:
(generally, indent by 2 spaces)